src/atoms/forms/canvas-sketch/react-canvas-sketch.tsx

```diff
@@ -0,0 +1,219 @@
+    onAddedStroke: (strokeSettings: CanvasDrawToolSettings) => void;
+    redrawIncrement: number;
+    canvasToolType: CanvasToolType;
+    showCanvasBorder: boolean;
+    toolWidth: number;
+    toolColor: string;
+    value: ReactCanvasSketchValue;
+}
+
+// #endregion Interfaces
+
+
+// -------------------------------------------------------------------------------------------------
+// #region Component
+// -------------------------------------------------------------------------------------------------
+
+const ReactCanvasSketch: React.FunctionComponent<ReactCanvasSketchProps> = (
+    props: React.PropsWithChildren<ReactCanvasSketchProps>
+) => {
+
+    // track the ref of the canvas elements
+    const htmlCanvasBackgroundImage = React.createRef<HTMLCanvasElement>();
+    const htmlCanvasSketch = React.createRef<HTMLCanvasElement>();
+
+    // track the previous list of objects
+    const prevValueObjectsRef = React.useRef<CanvasDrawToolSettings[]>();
+    useEffect(() => {
+        prevValueObjectsRef.current = props.value.objects;
+    });
+    const prevValueObjects = prevValueObjectsRef.current;
+
+    // initialize state
+    const canvasSketchDefaultInstance: CanvasSketch = null as any;
+    const [isInitialized, setIsInitialized] = React.useState(false);
+    const [canvasSketch, setCanvasSketch] = React.useState(canvasSketchDefaultInstance);
+
+    // set up effects
+    useEffect(() => {
+        console.log("useEffect: initializing");
+        if (!isInitialized) {
+            // set up the default options for the background image
+            const canvasSketchConfig: CanvasSketchConfig = {
+                backgroundImage: { src: props.backgroundImageUrl },
+                backgroundImageCanvas: htmlCanvasBackgroundImage.current as HTMLCanvasElement,
+                currentObjectIndex: props.value.currentObjectIndex,
+                objectStack: props.value.objects,
+                panX: 0,
+                panY: 0,
+                scaleFactor: 1,
+                sketchCanvas: htmlCanvasSketch.current as HTMLCanvasElement,
+            };
+
+            // initialize the canvas sketch object
+            const newCanvasSketch = new CanvasSketch(canvasSketchConfig);
+
+            // set up any default property driven settings
+            newCanvasSketch.setSelectedTool(props.canvasToolType);
+            newCanvasSketch.setToolColor(props.toolColor);
+            newCanvasSketch.setToolWidth(props.toolWidth);
+            newCanvasSketch.setOnAddedToolStroke(props.onAddedStroke);
+
+            // set state
+            setCanvasSketch(newCanvasSketch);
+            setIsInitialized(true);
+        }
+    }, [
+        canvasSketch,
+        htmlCanvasBackgroundImage,
+        htmlCanvasSketch,
+        isInitialized,
+        props.backgroundImageUrl,
+        props.canvasToolType,
+        props.onAddedStroke,
+        props.toolColor,
+        props.toolWidth,
+        props.value.currentObjectIndex,
```

Comment:
Just a note, we've experienced some weirdness with putting deeply nested properties in a dependency list for a hook. If it's working for you, great - just something that I have avoided due to some unexpected behavior.

Reply:
Honestly I wasn't planning on adding all of these dependencies but Typescript is configured to complain about it because they're all being used inside the hook.  In it's current implementation, all of these dependencies are necessary for configuring the `CanvasSketch` object at initialization, although I suppose `setSelectedTool()`, `setToolColor()`, `setToolWidth()`, and `setOnAddedToolStroke()` are probably not needed here since they're already being done in different hooks.  I'll remove them and validate that.  I'm all ears if you have any suggestions besides that.

Reply:
You could probably listen for `props` and then destructure the properties you care about inside of the hook - or, destructure the properties you care about from outside the hook, and then add those in the dependency list instead. 

Take a look at this thread: https://github.com/facebook/react/issues/16265

Reply:
@brandongregoryscott That thread was helpful, thanks!  I updated the code since it wasn't optimized for initialization of the canvas sketch so I got around it that way.  Nice to know to destructure in advance though according to best practive.